components/sidebar/sidebar_service.cc

```diff
@@ -72,6 +74,13 @@ SidebarItem::BuiltInItemType GetBuiltInItemTypeForLegacyURL(
   return SidebarItem::BuiltInItemType::kNone;
 }
 
+size_t GetLastBuiltInItemIndex(const std::vector<SidebarItem>& items) {
+  const auto it =
+      std::find_if(items.rbegin(), items.rend(),
+                   [](const SidebarItem& item) { return IsBuiltInType(item); });
+  return items.rend() - it;
```

Comment:
Is this going to end up being index `0` if no built-in items are found? I think that's what we want.

Reply:
Yes, if no built-in items, the result is `0`